src/index.js

```diff
@@ -195,6 +197,25 @@ export default class ToggleBlock {
     }
   }
 
+  /**
+   * With the key combination shift + tab the toggle block is extracted.
+   *
+   * @param {block} item
+   * @param {KeyboardEvent} e
+  */
+  extractionBlock(item, e) {
+    if (e.code === 'Tab' && e.shiftKey) {
+      const indexBlock = this.api.blocks.getCurrentBlockIndex();
+
+      const children = document.querySelectorAll(`div[foreignKey="${this.wrapper.id}"]`);
+      const length = children.length;
+
+      this.api.blocks.delete(indexBlock);
+
+      this.api.blocks.insert('paragraph', { text: item.textContent }, {}, indexBlock + length, true);
```

Comment:
@Jenietoc when there're multiple toggle blocks, for example, we have two toggle blocks (toggle A with 3 blocks inside it; toggle B with 2 blocks inside it), if you type `Shift + Tab` in the first block of the toggle A, that block will move inside of toggle B. 

To fix it you could change the index with which you're inserting the block to `indexBlock + length - 1`.

Reply:
I think you I generating more than one bug, and this solution would not be the most appropriate. let me think about it a little more. maybe you know how to count the number of blocks inside the editor, example: a toggle would be a block, creating a paragraph would be another block. and actually we would have two blocks, do you know how to do something like that?

Reply:
@Jenietoc Yes, you're right. I tested it again and that solution isn't working well either. To calculate the number of blocks in the editor you could use `this.api.blocks.getBlocksCount()`, but you should also consider that there might be more blocks after the toggle and its nested blocks.